.rubocop.yml

```diff
@@ -1,5 +1,5 @@
 AllCops:
-  TargetRubyVersion: 2.1
+  TargetRubyVersion: 2.6
```

Comment:
Can we set this TargetRubyVersion to 3.1 ?

Reply:
Tried changing TargetRubyVersion: 3.1 and getting a message as follows. If we want to use 3.1 as TargetRubyVersion then need to change gemspec as well.
```
[cloudservicesdev|hosting-dev:pc-team-23] ~/Documents/OfficeWorks/Projects/AcquiaProjects/e_microbus_2$ rake rubocop
....................................................................
features/fixtures/basic-app/basic.gemspec:14:29: C: Gemspec/RequiredRubyVersion: required_ruby_version and TargetRubyVersion (3.1, which may be specified in .rubocop.yml) should be equal.
  s.required_ruby_version = '>= 2.6'
                            ^^^^^^^^
microbus.gemspec:19:29: C: Gemspec/RequiredRubyVersion: required_ruby_version and TargetRubyVersion (3.1, which may be specified in .rubocop.yml) should be equal.
  s.required_ruby_version = '>= 2.6'
                            ^^^^^^^^

16 files inspected, 2 offenses detected
```
Tested with 3.1, in all these files and working fine.

Reply:
What is the goal of this PR? if the idea is to update it to ruby 3.1 then we could update gemspec too no?

Reply:
Since We are still using ruby versions lower than 3.1, seems like we are unable to change these for now.
```
[cloudservicesdev|hosting-dev:pc-team-23] ~/Documents/OfficeWorks/Projects/AcquiaProjects/e_microbus_2$ rvm list
   ruby-2.4.5 [ broken ]
   ruby-2.6.3 [ broken ]
   ruby-2.6.5 [ broken ]
   ruby-2.6.6 [ broken ]
=> ruby-2.7.0 [ broken ]
   ruby-2.7.3 [ arm64 ]
   ruby-2.7.5 [ arm64 ]
   ruby-3.1.0 [ arm64 ]
   ruby-3.1.2 [ arm64 ]

# => - current
# =* - current && default
#  * - default

[cloudservicesdev|hosting-dev:pc-team-23] ~/Documents/OfficeWorks/Projects/AcquiaProjects/e_microbus_2$ 
[cloudservicesdev|hosting-dev:pc-team-23] ~/Documents/OfficeWorks/Projects/AcquiaProjects/e_microbus_2$ bundle install
Warning: the running version of Bundler (2.1.2) is older than the version that created the lockfile (2.4.19). We suggest you to upgrade to the version that created the lockfile by running `gem install bundler:2.4.19`.
Fetching gem metadata from https://rubygems.org/.........
microbus-0.0.1 requires ruby version >= 3.1, which is incompatible with the current version, ruby 2.7.0p0
[cloudservicesdev|hosting-dev:pc-team-23] ~/Documents/OfficeWorks/Projects/AcquiaProjects/e_microbus_2$
```
Tested with this PR
https://github.com/acquia/microbus/pull/36/commits - 0eae4fc5a2fc5259ad68e680c21561f5c8fc5871
https://github.com/acquia/ah-web-server/pull/555 - a958fcf6cceece3423ebc570dcb2cd1ff640c1bf
https://core.cloudbees.ais.acquia.io/devops-pipeline-2-jenkins/blue/organizations/jenkins/AH-web-server-PIPELINE/detail/PR-555/4/pipeline/151
```
[2023-09-13T04:41:56.946Z] + ssh-agent sh -c echo $SSH_KEY | base64 -d | ssh-add - ; make package

[2023-09-13T04:41:56.954Z] Identity added: (stdin) ((stdin))

[2023-09-13T04:41:56.954Z] ./debian/build.sh

[2023-09-13T04:41:57.514Z] Don't run Bundler as root. Installing your bundle as root will break this

[2023-09-13T04:41:57.522Z] application for all non-root users on this machine.

[2023-09-13T04:42:07.447Z] microbus-0.0.1 requires ruby version >= 3.1, which is incompatible with the

[2023-09-13T04:42:07.454Z] current version, 2.7.4

[2023-09-13T04:42:07.454Z] make: *** [Makefile:37: package] Error 5

script returned exit code 2
```

Reply:
There is a dependency on other repos from multiple teams.. once they upgrade it to the Ruby 3.1.2 then we can update here as well. ... but the goal of this PR is to make other changes to support ruby 3.1.2 when other teams update their repos with the same.

Reply:
Created this ticket to address this, once this dependency is resolved - [PC-17326](https://backlog.acquia.com/browse/PC-17326)